packages/sanity/src/core/util/draftUtils.ts

```diff
@@ -210,3 +210,16 @@ export function removeDupes(documents: SanityDocumentLike[]): SanityDocumentLike
     .map((entry) => entry.draft || entry.published)
     .filter(isNonNullable)
 }
+
+/**
+ * @beta
+ * Given a document id, it indicates whether it is a draft, version or published document.
+ */
+export type DocumentVariantType = 'draft' | 'version' | 'published'
+
+/** @beta */
+export function getDocumentVariantType(documentId: string): 'draft' | 'version' | 'published' {
+  if (isDraftId(documentId)) return 'draft'
+  if (isVersionId(documentId)) return 'version'
+  return 'published'
```

Comment:
Not strictly necessarily for approval, but Since it's a `@beta` we can probably give it just a bit more context in the comments generically about what it does and returns :) the description you have on this PR is pretty good so I would just adapt it to here!